### PR TITLE
Online workstation

### PR DESCRIPTION
**Pull request by rayrayraykk**
You must set the following vars in .env:
```
SECRET_KEY=****
SESSION_TYPE==****
IP==****
PORT==****
COPILOT_IP==****
COPILOT_PORT==****
CLIENT_ID==****
OWNER==****
REPO==****
USER_FILE_NAME==****
OSS_ENDPOINT==****
OSS_BUCKET_NAME==****
OSS_ACCESS_KEY_ID==****
OSS_ACCESS_KEY_SECRET==****
CLIENT_SECRET==****
LOCAL_WORKSTATION==****
```

## Checklist

Please check the following items before code is ready to be reviewed.

- [X]  Code has passed all tests
- [X]  Docstrings have been added/updated in Google Style
- [X]  Documentation has been updated
- [X]  Code is ready for review

### CODE REVIEW

**Review by qbc2016 — COMMENTED** [review]
LGTM.

File: src/agentscope/studio/_app_online.py [hunk 1]
```diff
@@ -0,0 +1,397 @@
...
+    if lang == "zh":
+        refresh()
+        response.set_cookie("locale", "zh")
+        return response
+
+    return jsonify({"data": "success"})
+
+
+if __name__ == "__main__":
+    _app.run(host="0.0.0.0", port=PORT)
```
Comment (pan-x-c) [inline src/agentscope/studio/_app_online.py:397]: make this configurable

File: setup.py [hunk 2]
```diff
@@ -126,6 +133,7 @@
         "distribute": distribute_requires,
         "dev": dev_requires,
         "full": full_requires,
+        "online": online_requires,
```
Comment (pan-x-c) [inline setup.py:136]: add `online_requires` into `full_requires`

**Review by pan-x-c — COMMENTED** [review]
Please see the inline comments, others LGTM

File: setup.py (hunk 2 above)
Comment (DavdGao) [reply setup.py:136]: I agree with weirui, and we shouldn't expose the online requires to users. 
Plz remove it from `extras_require`

File: src/agentscope/studio/utils.py [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+from functools import wraps
+from typing import Any, Callable
+
+import jwt
+from flask import session, redirect, url_for, abort
+
+EXP_TIME = 1440  # One day long
+
+
+def require_auth(
```
Comment (DavdGao) [inline src/agentscope/studio/utils.py:20]: Prefix all these functions with "_" to avoid exposing to users.

**Review by DavdGao — COMMENTED** [review]
Please see inline comments

**Review by qbc2016 — APPROVED** [review]
LGTM.

**Pull request merged**